common/util/additionalArguments.go

```diff
@@ -0,0 +1,47 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+)
+
+// Gets allowed additional argument types and indicates whether it is an action argument. Also returns if there is an error or not.
+func GetAllowedAdditionalArgumentsForType(typ string) (map[string]bool, error) {
+	switch typ {
+	case common.DATABASE_TYPE_MSSQL:
+		return map[string]bool{
```

Comment:
these seem like static maps. can we define it in the beginning as constants and return them here?
also, @manavrajvanshi few months later, if these maps change in a new version of NDB, do we have to watch out for anything right now so there are no issues later?

Reply:
This make sense, will address shortly. Also, if these maps change and we decide to use a new NDB api version, then yes, we would have to verify that the additionalArguments are being passed in the appropriate provisioning body property (which in this case is currently all in the actionArguments property). I don't think this is unique to this PR change, we would always have to do this I believe. Is that correct @manavrajvanshi?